File: app/src/main/java/com/fastaccess/provider/timeline/handler/LinkHandler.java
package com.fastaccess.provider.timeline.handler;

import android.graphics.Color;
import android.text.SpannableStringBuilder;

import com.zzhoujay.markdown.style.LinkSpan;

import net.nightwhistler.htmlspanner.TagNodeHandler;

import org.htmlcleaner.TagNode;

/**
 * Created by Kosh on 10 May 2017, 8:46 PM
 */

public class LinkHandler extends TagNodeHandler {
    private final static int linkColor = Color.parseColor("#4078C0");

    @Override public void handleTagNode(TagNode node, SpannableStringBuilder spannableStringBuilder, int start, int end) {
        String href = node.getAttributeByName("href");
        if (href != null) {
            spannableStringBuilder.setSpan(new LinkSpan(href, linkColor), start, end, 33);
        } else if (node.getText() != null) {
            spannableStringBuilder.setSpan(new LinkSpan("https://github.com/" + node.getText().toString(), linkColor), start, end, 33);
        }
    }
}

File: markdown/src/main/java/com/fastaccess/markdown/spans/LinkHandler.kt
package com.fastaccess.markdown.spans

import android.graphics.Color
import android.text.SpannableStringBuilder
import net.nightwhistler.htmlspanner.SpanStack
import net.nightwhistler.htmlspanner.TagNodeHandler
import org.htmlcleaner.TagNode

/**
 * Created by Kosh on 10 May 2017, 8:46 PM
 */

class LinkHandler : TagNodeHandler() {

    override fun handleTagNode(
        node: TagNode,
        builder: SpannableStringBuilder,
        start: Int,
        end: Int,
        spanStack: SpanStack
    ) {
        val href = node.getAttributeByName("href")
        val url = if (!href.isNullOrEmpty()) {
            href.toString()
        } else if (!node.text.isNullOrEmpty()) {
            "https://github.com/${node.text}"
        } else {
            null
        }

        url?.let { builder.setSpan(UrlSpan(href, linkColor), start, end, 33) }
    }

    companion object {
        private val linkColor = Color.parseColor("#4078C0")
    }
}
